### PR TITLE
feat!: bump fuel core to 0.10.1

### PR DESCRIPTION
**Pull request by hal3e**
closes: https://github.com/FuelLabs/fuels-rs/issues/537

There are still three test that fail. Other then that I have:
- Removed `byte_price` from TxParameters
- Updated estimation of transaction cost
- Added TxPointer to UTXOs

WASM test are also failing I will disable them temporarily.

### CODE REVIEW

**Review by Voxelot — DISMISSED** [review]

**Review by adlerjohn — COMMENTED** [review]
merge conflicts

File: packages/fuels-types/src/lib.rs [hunk 1]
```diff
@@ -124,7 +124,7 @@ pub struct TypeDeclaration {
     pub type_parameters: Option<Vec<usize>>,
 }
 
-#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[derive(Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
```
Comment (MujkicA) [inline packages/fuels-types/src/lib.rs:127]: I assume this is related to the same clippy issue we discussed

**Review by iqdecay — APPROVED** [review]
LGTM! :ship:

File: examples/contracts/src/lib.rs [hunk 2]
```diff
@@ -320,14 +314,15 @@ mod tests {
         // ANCHOR_END: simulate
         let response = contract_instance.mint_coins(1_000_000).call().await?;
         // ANCHOR: variable_outputs
-        let address = wallet.address();
+        // TODO: Enable test
```
Comment (MujkicA) [inline examples/contracts/src/lib.rs:317]: Could you elaborate on why these tests are failing?

File: packages/fuels-signers/src/provider.rs [hunk 3]
```diff
@@ -503,26 +494,31 @@ impl Provider {
...
+                .expect("Error calculating TransactionFee");
+
+        Ok(TransactionCost {
+            min_gas_price: min_gas_price.0,
+            gas_price,
             gas_used,
-        )
+            metered_bytes_size: tx.metered_bytes_size() as u64,
+            total_fee: transaction_fee.total(),
+        })
```
Comment (digorithm) [inline packages/fuels-signers/src/provider.rs:521]: This is awesome!

**Review by digorithm — APPROVED** [review]
Nice -- we will need to urgently re-enable these tests (and add proper support for them to work) once we have the blocker gone.

For clarity, in the upcoming release, if contracts are using either of these:

```
mint_to_address(amount_to_mint, recipient);
transfer_to_output(amount_to_transfer, ~ContractId::from(BASE_TOKEN), recipient);
msg_sender();
```

The SDK will not work properly. 

These will work again in the subsequent release after the Sway/stdlib work is done to unblock this.

File: examples/contracts/src/lib.rs (hunk 2 above)
Comment (digorithm) [reply examples/contracts/src/lib.rs:317]: @MujkicA https://github.com/FuelLabs/fuels-rs/pull/546#pullrequestreview-1089040916 should answer! TLDR: stdlib needs to support some new stuff.

**Review by iqdecay — APPROVED** [review]
lgtm

**Pull request merged**